python/anvill/binja.py

```diff
@@ -266,7 +321,7 @@ def get_type(ty):
         return ty.type()
 
     elif isinstance(ty, bn.Type):
-        return _convert_bn_type(ty, {})
+        return _convert_bn_type(bv, ty, {})
```

Comment:
Instead of using an empty dict here for the type cache, can you have `BNProgram` have a private, initially empty dictionary that is passed around? This means modifiying `get_type` to take in another parameter, or to take in the `BNProgram` and get at the `bv` through it.

Reply:
Thanks, @pgoodman for the suggestion. I would like to have this change in a separate PR. The changes are not many but It will be cleaner to have it that way. Let me know how do you think?

Reply:
I agree, in a separate PR, maybe make a class like `TypeLifter` or `TypeCache`, where the cache itself has methods to convert each binary ninja type, and internally manages a cache.